### ISSUE TITLE
[Bug]: Laravel 11.17.0 added `HasCollection` trait and new `$collectionClass` property

### ISSUE THREAD

**Issue by dmason30** [laravel-idea/plugin#1035]
### Bug description

See these PRs the `HasCollection` trait and `$collectionClass` property was added so we can properly type hint the model collection for static-analysis (e.g. PHPStan). 

https://github.com/laravel/framework/pull/52171
https://github.com/laravel/framework/pull/52186

So now we have this on a Model:

```php
class Post extends Model
{
    /** @use HasCollection<PostCollection<array-key, static>> */
    use HasCollection;

    protected static string $collectionClass = PostCollection::class;
}
```

### Plugin version

8.2.2.241

### Operating system

MacOS

### Steps to reproduce

Use the code above and the plugin when running `Generate Helper Code` does not pickup the custom collections.

**Comment by adelf** [comment]
Well, marking this as a "bug" is a bit... early)

I'll try to use this declarations.

**Comment by dmason30** [comment]
Sorry if I chose the wrong type. Just didn't feel right as a feature request either. 

Thanks for looking into it 👍

**Comment by adelf** [comment]
Implemented in the Laravel Idea 8.2.3
